c10/core/InferenceMode.h

```diff
@@ -60,9 +60,9 @@ struct TORCH_API InferenceMode {
     DispatchKeySet excluded = enabled
         ? (prev_keyset.excluded_ | c10::autograd_dispatch_keyset)
         : (prev_keyset.excluded_ - c10::autograd_dispatch_keyset);
-    c10::impl::PODLocalDispatchKeySet cur_keyset;
-    cur_keyset.set_included(included);
-    cur_keyset.set_excluded(excluded);
+    auto cur_keyset = prev_keyset;
+    cur_keyset.included_ = included;
+    cur_keyset.excluded_ = excluded;
```

Comment:
Note that what's happening here is `cur_keyset` is being initialized from `prev_keyset`, then both (i.e., all) of its properties are being updated. Adding another ctor to `LocalDispatchKeySet` which took DispatchKeySets for `included_` and `excluded_` directly would remove the need for them to be non-const here. (Don't know if there are other places but I imagine they could be dealt with similarly.)

At a high level, the pitch is that by making `LocalDispatchKeySet` axiomatically immutable, and making (the class currently known as) `LocalDispatchKeySetWrapper` axiomatically a facade on TLS, everything's a bit cleaner and easier to reason about - e.g. you don't need to work as hard to figure out whether a property update is gonna trickle back into TLS or not.